intern/spindle/SpindleEncryption.cpp

```diff
@@ -0,0 +1,301 @@
+			while (ii < t)
+			{
+				data[ii] -= h + (((char)offset)|((char)ii));//char(ii) - h + (char(offset)|char(ii));
+				ii++;
+			}
+		}
+		chunkSize = pieceSize;
+	}
+}
+
+static void SpinEncrypt_Hex_64(char* data, int dataSize, char* key)
+{
+	int keySize = 0, i;
+	unsigned long long realKey = 0, s;
+   if (key == NULL)
+      return;
+	keySize = strlenN(key);
+	for (i = 0; i < keySize; i++)
+	{
+		s = keySize - 1 - i;
+		if ((key[i] >= '0')&&(key[i] <= '9'))
+			realKey += ((unsigned long long)(key[i]-'0')<<(s<<2));
+		else if ((key[i] >= 'a')&&(key[i] <= 'f'))
+			realKey += ((unsigned long long)(key[i]-'a'+10)<<(s<<2));
+		else if ((key[i] >= 'A')&&(key[i] <= 'F'))
+			realKey += ((unsigned long long)(key[i]-'A'+10)<<(s<<2));
+		else
+			realKey += ((unsigned long long)(key[i])<<(s<<2));
+	}
+	SpinEncrypt(data, dataSize, realKey);
+}
+
+static void SpinDecrypt_Hex_64(char* data, int dataSize, char* key)
+{
+	int keySize = 0, i;
+	unsigned long long realKey = 0, s;
+   if (key == NULL)
+      return;
+	keySize = strlenN(key);
+	for (i = 0; i < keySize; i++)
+	{
+		s = keySize - 1 - i;
+		if ((key[i] >= '0')&&(key[i] <= '9'))
+			realKey += ((unsigned long long)(key[i]-'0')<<(s<<2));
+		else if ((key[i] >= 'a')&&(key[i] <= 'f'))
+			realKey += ((unsigned long long)(key[i]-'a'+10)<<(s<<2));
+		else if ((key[i] >= 'A')&&(key[i] <= 'F'))
+			realKey += ((unsigned long long)(key[i]-'A'+10)<<(s<<2));
+		else
+			realKey += ((unsigned long long)(key[i])<<(s<<2));
+	}
+	SpinDecrypt(data, dataSize, realKey);
+}
+
+
+void SpinEncrypt_Hex(char* data, int dataSize, char* key)
+{
+	int keySize = 0, charPos, i;
+	if (key == NULL)
+      return;
+   keySize = strlenN(key);
+   charPos = keySize - (keySize % 16);
+	if (keySize == charPos)
+		charPos -= 16;
+	if (keySize <= 16)
+		SpinEncrypt_Hex_64(data, dataSize, key);
+	else
+	{
+		char tempKey[17];
+		tempKey[16] = 0;
+		while (charPos >= 0)
+		{
+			for (i = 0; ((i < 16)&&(i < keySize)); i++)
+				tempKey[i] = key[i+charPos];
+			charPos -= 16;
+			SpinEncrypt_Hex_64(data, dataSize, tempKey);
+		}
+	}
+}
+
+void SpinDecrypt_Hex(char* data, int dataSize, char* key)
+{
+	int keySize = 0, charPos = 0, i;
+	if (key == NULL)
+      return;
+   keySize = strlenN(key);
+   if (keySize <= 16)
+		SpinDecrypt_Hex_64(data, dataSize, key);
+	else
+	{
+		char tempKey[17];
+		tempKey[16] = 0;
+		while (charPos < keySize)
+		{
+			for (i = 0; ((i < 16)&&(i < keySize)); i++)
+				tempKey[i] = key[i+charPos];
+			charPos += 16;
+			SpinDecrypt_Hex_64(data, dataSize, tempKey);
+		}
+	}
+}
+
+void SpinEncryption_FindAndSet_Key(char **argv, int i, char *&hexKey)
+{
+	/* Find main key */
+	int hexStrSize = 0, argPos = 2, maxStringLen = int(strlen(argv[i]));
+
+	for (hexStrSize = 0; ((argv[i][hexStrSize + argPos] != 0) && (argv[i][hexStrSize + argPos] != '.')); hexStrSize++){}
+
+	hexKey = new char[hexStrSize + 1];
```

Comment:
considering `char*` is a type, when you change it's value nothing happen for the caller function if the variable isn't a reference or a pointer to.

Here i think the easiest way would be using a double pointer:
```… char **hexKey)
*hexKey = new char hexKey```

Maybe stupid question: why don't you return the hexKey ?